app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/SelectionInteractionViewModel.kt

```diff
@@ -67,6 +75,18 @@ class SelectionInteractionViewModel private constructor(
           )
         }
       }
+    if (selectedAnswer.size == 1 && isSubmitAnswerEnabled) {
+      val selectedIndex = selectedAnswer[0]
+      val isAnswerUpdated =
+        updateSelection(selectedIndex, choiceItems[selectedIndex].isAnswerSelected.get())
+      choiceItems[selectedIndex].isAnswerSelected.set(isAnswerUpdated)
+    } else if (selectedAnswer.size > 1 && isSubmitAnswerEnabled) {
```

Comment:
Why is isSubmitAnswerEnabled needed? We should have all the info already in this view model for this--isSubmitAnswerEnabled is false if there's a runtime error (or a submit-time error discovered after trying to submit the answer). Piping this here sort of creates a logical circular dependency which makes things a bit confusing.

Reply:
Is this explanation still valid? It doesn't seem like this approach is being used in the latest version.

Reply:
Yes, now as we are resetting the answer on submit button clicked, we don't need to check this condition.